### ISSUE TITLE
[Bug]: Cannot find module 'storybook/internal/common'

### ISSUE THREAD

**Issue by eavam** [storybookjs/test-runner#505]
### Describe the bug

Running `yarn test-storybook` responds with an error `Cannot find module 'storybook/internal/common'`.
On version 0.16 everything is fine. With 0.17 the problems start.

### To Reproduce

```
node:internal/modules/cjs/loader:1147
  throw err;
  ^

Error: Cannot find module 'storybook/internal/common'
Require stack:
- <-- PATH -->/node_modules/@storybook/test-runner/node_modules/@storybook/core-common/shim.js
- <-- PATH -->/node_modules/@storybook/test-runner/dist/test-storybook.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
    at Module._load (node:internal/modules/cjs/loader:985:27)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (<-- PATH -->/node_modules/@storybook/test-runner/node_modules/@storybook/core-common/shim.js:1:18)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '<-- PATH -->/node_modules/@storybook/test-runner/node_modules/@storybook/core-common/shim.js',
    '<-- PATH -->/node_modules/@storybook/test-runner/dist/test-storybook.js'
  ]
}
```

### System

```bash
System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M1 Max
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
    Yarn: 4.1.1 - ~/.nvm/versions/node/v20.10.0/bin/yarn <----- active
    npm: 10.2.3 - ~/.nvm/versions/node/v20.10.0/bin/npm
  Browsers:
    Chrome: 127.0.6533.120
    Safari: 17.5
  npmPackages:
    @storybook/addon-essentials: 8.1.1 => 8.1.1 
    @storybook/addon-interactions: 8.1.1 => 8.1.1 
    @storybook/addon-links: 8.1.1 => 8.1.1 
    @storybook/blocks: 8.1.1 => 8.1.1 
    @storybook/preview-api: 8.1.1 => 8.1.1 
    @storybook/react: 8.1.1 => 8.1.1 
    @storybook/react-vite: 8.1.1 => 8.1.1 
    @storybook/test: 8.1.1 => 8.1.1 
    @storybook/test-runner: 0.17.0 => 0.17.0 
    eslint-plugin-storybook: 0.8.0 => 0.8.0 
    msw-storybook-addon: 2.0.3 => 2.0.3 
    storybook: 8.1.1 => 8.1.1 
    storybook-addon-mock: 5.0.0 => 5.0.0
```

**Comment by kumarbha** [comment]
I am also facing the same issue. Pls help

**Comment by bklatka** [comment]
Facing same issue as above

**Comment by dudiharush** [comment]
Me too...

**Comment by dudiharush** [comment]
This bug seems to be resolved after running: `npx storybook@8.2.0 upgrade` @eavam @kumarbha

**Comment by SoaresMG** [comment]
Same here, but I'm using storybook@7.6.19 with @storybook/test-runner@0.17.0 (I have also tried 0.19.1).

I did upgrade other deps that's when the issue started happening, I'm trying to find which one caused it

**Comment by ericsoderberghp** [comment]
FYI, I've found that if I remove `@storybook/addon-designs` then the error goes away and I am able to run.

**Comment by SoaresMG** [comment]
> Same here, but I'm using storybook@7.6.19 with @storybook/test-runner@0.17.0 (I have also tried 0.19.1).
> 
> 
> 
> I did upgrade other deps that's when the issue started happening, I'm trying to find which one caused it

It seems that my issue was related with being in storybook@7 using test-runner@0.17.0 which had storybook@8 deps.

I switched to 0.16.0 and it started working again.

It was working before with storybook@7 due to the magic hand of the lock file god, that allowed some storybook@8.0.5 deps to be mixed with storybook@7 itself (if I'm not mistaken it was some deps like @storybook/channels and @storybook/types and 1-2 more).

**Comment by sunilrathore24** [comment]
Same here - 

![image](https://github.com/user-attachments/assets/25c22c54-1f95-4d9a-97de-d592d807e21b)

```
  "devDependencies": {
    "@angular-devkit/build-angular": "18.0.2",
    "@angular-devkit/core": "18.0.2",
    "@angular-devkit/schematics": "18.0.2",
    "@angular-eslint/eslint-plugin": "18.0.1",
    "@angular-eslint/eslint-plugin-template": "18.0.1",
    "@angular-eslint/template-parser": "18.0.1",
    "@angular/cli": "~18.0.0",
    "@angular/compiler-cli": "18.0.1",
    "@angular/language-service": "18.0.1",
    "@babel/core": "^7.24.7",
    "@babel/plugin-proposal-decorators": "^7.24.7",
    "@babel/preset-env": "^7.24.7",
    "@babel/preset-typescript": "^7.24.7",
    "@commitlint/cli": "^17.0.2",
    "@commitlint/config-conventional": "^17.0.2",
    "@commitlint/prompt-cli": "^17.0.0",
    "@custom-elements-manifest/analyzer": "~0.5.7",
    "@mjhenkes/postcss-rtl": "^2.0.0",
    "@nx/cypress": "19.3.2",
    "@nx/eslint": "19.3.2",
    "@nx/eslint-plugin": "19.3.2",
    "@nx/js": "19.3.2",
    "@nx/storybook": "19.3.2",
    "@nx/workspace": "19.3.2",
    "@open-wc/testing": "^3.1.6",
    "@rollup/plugin-alias": "^5.1.0",
    "@rollup/plugin-node-resolve": "^15.2.3",
    "@rollup/plugin-terser": "^0.4.4",
    "@rollup/plugin-typescript": "^11.1.6",
    "@rollup/plugin-virtual": "^3.0.2",
    "@schematics/angular": "18.0.2",
    "@storybook/addon-a11y": "^8.1.7",
    "@storybook/addon-essentials": "^8.1.7",
    "@storybook/addon-webpack5-compiler-babel": "^3.0.3",
    "@storybook/angular": "^8.1.7",
    "@storybook/blocks": "^8.1.7",
    "@storybook/components": "^8.1.7",
    "@storybook/core-events": "^8.1.7",
    "@storybook/core-server": "^8.1.7",
    "@storybook/manager-api": "^8.1.7",
    "@storybook/preview-api": "^8.1.7",
    "@storybook/test-runner": "^0.19.1",
    "@storybook/theming": "^8.1.7",
    "@storybook/types": "^8.1.7",
    "@storybook/web-components": "^8.1.7",
    "@storybook/web-components-webpack5": "^8.1.7" }
```

**Comment by ekeijl** [comment]
Running into the same issue when using Yarn PnP:

```
> yarn run storybook-test

/<REDACTED>/.pnp.cjs:67456
      Error.captureStackTrace(firstError);
            ^

Error: @storybook/core-common tried to access storybook (a peer dependency) but it isn't provided by its ancestors; this makes the require call ambiguous and unsound.

Required package: storybook (via "storybook/internal/common")
Required by: @storybook/core-common@virtual:fa7a63bce88eef2cb8a3f1c5ece91110f0f08289f1e8517bf01d3d69cbeef0a9a7bfcaa746c2f0e178177003417e002337645312a23a1888f0c1e9c5fa39c856#npm:8.2.9 (via /<REDACTED>/.yarn/__virtual__/@storybook-core-common-virtual-14814bf21c/0/cache/@storybook-core-common-npm-8.2.9-cf49c5e44e-0d70bd588a.zip/node_modules/@storybook/core-common/)

Ancestor breaking the chain: @storybook/test-runner@npm:0.19.1
```

It seems the `@storybook/test-runner` package is missing a `peerDependency` on the `storybook` package?

As a workaround, this can be patched with Yarn in your `.yarnrc.yml` ([docs](https://yarnpkg.com/configuration/yarnrc#packageExtensions)):

```yml
packageExtensions:
   '@storybook/test-runner@*':
      peerDependencies:
         storybook: ">=8"
```

Make sure to install the `storybook` package as a dependency in your project.

**Comment by yannbf** [comment]
Hey everyone! Thanks for reporting this.

Here is a quick summary:
- This issue is the result of a structural change that was done in Storybook 8.2, which affects the way packages are requested from other dependencies.
- This issue only happens if you are using Storybook 8 dependencies, could be coming from an addon like `@storybook/addon-designs` for instance.
- If you are using Storybook 7, you must stay at version 0.16.0 of the test-runner, as stated in [this table](https://github.com/storybookjs/test-runner#storybook-compatibility).
- [I have a PR](https://github.com/storybookjs/test-runner/pull/512) that updates the test-runner itself to follow the new structure of packages from Storybook, which should resolve this issue. This should be fixed in the next minor release of the test-runner.

**Comment by daaashley** [comment]
I spent a few weeks debugging this same error message when upgrading from v7 to v8, so I thought I would share my scenario to maybe help others save time. If you've come across this page/error and already tried everything else, theres a chance your issue might be similar. In my case, I was using storybook in a turbo monorepo. In the monorepo root package.json I declared two npm workspaces 
```
"workspaces": [
    "apps/*",
    "packages/*"
  ],
```
my storybook application was located at 'apps/storybook'. If you are using an npm workspace, npm will create a copy of your workspace folder application under that applications package.json "name" key and place it in your root node_modules. If your packages.json "name" key is "storybook", then npm will copy your application into the root node_modules as 'storybook'. This creates a node_modules folder name conflict with the real 'storybook' dependency, and npm hoisting behaviour can become unpredictable and move the actual storybook dependency to a different node_modules folder in your monorepo because it treats it as a different version. I'm not sure if this applies to yarn or pnpm, but this could be something to look at.
